File: backend/app/schemas/lot_set.py
from datetime import datetime

from app.schemas.common import ApiModel


class LotSetOut(ApiModel):
    id: int
    name: str
    uploaded_at: datetime
    lots_count: int
    is_active: bool


class FeedUploadResult(ApiModel):
    set: LotSetOut
    skipped: int  # лоты фида, не прошедшие разбор

File: frontend/src/domain/lotSet.ts
/** Набор лотов — результат одной загрузки фида. Наборы immutable,
 *  публично виден только активный. */

export interface LotSet {
  id: number
  name: string
  uploadedAt: Date
  lotsCount: number
  isActive: boolean
}

export interface FeedUploadResult {
  set: LotSet
  /** Лоты фида, не прошедшие разбор — импорт из-за них не падает */
  skipped: number
}
